### PR TITLE
fix(wallet/contract): bounded updates: recent offers / payments only

### LINKED ISSUE #6065: smart wallet state grows with each offer
## Describe the bug

The published state includes a list of all offers, past and present.

## Expected behavior
that the offer is removed when:
- the offerResult does not support continuing offers (all PSM offers, but not vault creation)
- or the offerResult is an error (since you cannot continue that)
- then once the offer is exited, the state update is reported via subscription and the offer is deleted from the wallet

cc @warner @samsiegart @turadg @dtribble @michaelfig

### PR DESCRIPTION
**Pull request by dckc**
closes: #6065 

## Description

Rather than further work on pruning outdated state, the approach here is to notify only about the most recently updated payment or offer. We can get the rest from historical data.

There remains a case of batch-updates to offers. It wasn't a problem in a test of 50 trades, up to 15 at a time.

This also fixes a problem where every status update for a payment caused the record to get deeper and deeper.

The list of purses only grows via VBANK updates, so it's not likely to get out of hand.

### Security Considerations

none, I believe.

### Documentation Considerations

I updated the `smartWallet.js` notification API docs. Clients will have to know to look thru history.

[psm-tool.js](https://gist.github.com/dckc/dfc2182a59b11ec4527349afd5ca050f#file-psm-tool-js) is a compatible client.

### Testing Considerations

I did several integration test runs and updated relevant unit tests.

### CODE REVIEW

File: packages/wallet/api/src/lib-wallet.js [hunk 1]
```diff
@@ -523,7 +524,7 @@ export function makeWalletRoot({
     inboxState.set(id, offerForDisplay);
     if (doPush) {
       // Only trigger a state change if this was a single update.
-      offersUpdater.updateState([...inboxState.values()]);
+      offersUpdater.updateState([offerForDisplay]);
```
Comment (turadg) [inline packages/wallet/api/src/lib-wallet.js:527]: this doesn't help with `updateAllInboxState`. Missed or deferred?

File: packages/wallet/contract/src/smartWallet.js [hunk 2]
```diff
@@ -50,9 +50,13 @@ export const makeSmartWallet = async (
   const notifierParts = {
     contacts: E(admin).getContactsNotifier(),
     dapps: E(admin).getDappsNotifier(),
+    /** all current issuers */
     issuers: E(admin).getIssuersNotifier(),
+    /** recently updated offers */
```
Comment (turadg) [inline packages/wallet/contract/src/smartWallet.js]: "recently" seems unnecessarily vague. It's exactly the latest offer in some cases and it's all offers when `updateAllState` is called (whenever petnames are added or changed).

File: packages/wallet/contract/src/smartWallet.js [hunk 3]
```diff
@@ -50,9 +50,13 @@ export const makeSmartWallet = async (
...
     offers: E(admin).getOffersNotifier(),
+    /** recently updated payments */
```
Comment (turadg) [inline packages/wallet/contract/src/smartWallet.js]: singular "most recently payment". can still be an array for backwards compatibility and comment might explain that.

File: packages/wallet/api/src/lib-wallet.js (hunk 1 above)
Comment (dckc) [reply packages/wallet/api/src/lib-wallet.js:527]: um... deferred... as in... I'm not sure what that thing does... let's see if it's a problem in practice... moderate integration testing didn't turn up anything bad.

File: packages/wallet/contract/src/smartWallet.js (hunk 2 above)
Comment (dckc) [reply packages/wallet/contract/src/smartWallet.js]: ah. is that how it works? thanks!

**Review by turadg — APPROVED** [review]
LGTM

**Pull request merged**